lib/DBD/mysql.pm

```diff
@@ -1186,6 +1186,18 @@ flags:
 These are used to specify the respective parameters of a call
 to mysql_ssl_set, if mysql_ssl is turned on.
 
+=item mysql_ssl_verify_server_cert
+
+This turns on the MYSQL_OPT_SSL_VERIFY_SERVER_CERT connection option.
+
+Note that this is deprecated as of MySQL 5.7.11 and is removed in MySQL 8.0
+
+=item mysql_ssl_mode
+
+Can be set to 'disabled', 'preferred', 'required', 'verify_ca' or 'verify_identity'.
+Default is 'preferred'.
+
+Available from MySQL 5.7.11
```

Comment:
Thank you for investigation!

> 1. Add ssl_prefer. Keep disabled by default for backwards compatibility. Maybe change this to ssl_prefer=1 at a later stage?

Now I'm thinking if it is good idea to have enabled SSL by default. Basically lot of people are running MySQL either on localhost or in secure network where is encryption just wasting of resources. And those who needs encryption still explicitly request for it.

So I'm OK with `ssl_prefer=0` as default option.

Maybe I would suggest to throw error when both option `ssl` and `ssl_prefer` are specified.

> 2. Set SSL_MODE_VERIFY_CA if a ca file is specified (probably already what is happening now, but this could make it more explicit.

Yes make sense. If somebody specify certificate CA the only thing what can be done is to verify if CA matches.

> What should we expect to happen with MariaDB and mysql_ssl_preferred?

MariaDB does not really support this opportunistic encryption mode? In this case throw error. And probably report bug to MariaDB bug tracker.

> In 5.6 and older there was no real option to force the use of SSL/TLS on the client.

Then this should be in DBD::mysql documentation.


Anyway, I'm thinking if new attribute `ssl_prefer` is needed and if is better or even worse to introduce tri-state logic for `ssl`: `ssl=0` (always disabled; default), `ssl=1` (always enabled), `ssl=2` (prefer if available).